tests/test_cookiejar.py

```diff
@@ -253,6 +253,78 @@ async def test_filter_cookies_str_deprecated(loop: Any) -> None:
         jar.filter_cookies("http://éé.com")
 
 
+@pytest.mark.parametrize(
+    ("url", "expected_cookies"),
+    [
+        (
+            "http://pathtest.com/one/two/",
+            [
+                "path2-cookie",
+                "shared-cookie",
+                "path3-cookie",
+                "path4-cookie",
+            ],
+        ),
+        (
+            "http://test1.example.com/",
+            [
+                "shared-cookie",
+                "domain-cookie",
+                "subdomain1-cookie",
+                "dotted-domain-cookie",
+            ],
+        ),
+        (
+            "http://pathtest.com/",
+            [
+                "shared-cookie",
+                "no-path-cookie",
+                "path1-cookie",
+            ],
+        ),
+    ],
+    ids=(
+        "/one/two/ path",
+        "test1.example.com subdomain",
+        "pathtest.com",
+    ),
```

Comment:
This adds a couple more test cases, which includes the error I mentioned in the other thread (with /one/two vs /one/two/).

```suggestion
    (
        (
            "http://pathtest.com/one/two/",
            (
                "no-path-cookie",
                "path1-cookie",
                "path2-cookie",
                "shared-cookie",
                "path3-cookie",
                "path4-cookie",
            ),
        ),
        (
            "http://pathtest.com/one/two",
            (
                "no-path-cookie",
                "path1-cookie",
                "path2-cookie",
                "shared-cookie",
                "path3-cookie",
            ),
        ),
        (
            "http://pathtest.com/one/two/three/",
            (
                "no-path-cookie",
                "path1-cookie",
                "path2-cookie",
                "shared-cookie",
                "path3-cookie",
                "path4-cookie",
            ),
        ),
        (
            "http://test1.example.com/",
            (
                "shared-cookie",
                "domain-cookie",
                "subdomain1-cookie",
                "dotted-domain-cookie",
            ),
        ),
        (
            "http://pathtest.com/",
            (
                "shared-cookie",
                "no-path-cookie",
                "path1-cookie",
            ),
        ),
    ),
```